signaturehash.go

```diff
@@ -1,24 +1,42 @@
 package bt
 
 import (
+	"bytes"
 	"encoding/binary"
 	"errors"
 
 	"github.com/libsv/go-bk/crypto"
+	"github.com/libsv/go-bt/v2/bscript"
 	"github.com/libsv/go-bt/v2/sighash"
 )
 
+var defaultHex = []byte{1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
```

Comment:
what is this defaultHex for? @Tigh-Gherr

Reply:
Check the comment on line roughly line 160 in `signaturehash.go` https://github.com/libsv/go-bt/pull/34/files#diff-2db55662d065545f1027e43f3030a2bd58ac1776c1a69ddeddb296cd0d31cb81R160

Reply:
but it's only in the legacy signaturehash calculation isn't it? shouldn't we keep just in there instead of putting it in `CalcInputSignatureHash` as well?